### PR TITLE
Mesh generation

### PR DESCRIPTION
**Pull request by StephenRUK**
* Add command for mesh generation
* Writes OFF format meshes to a given output directory

Untested due to problems with importing the relative modules

TODO:
[ ] add skimage dependency to CI
[ ] add mesh tools dependency to CI

### CODE REVIEW

File: geocruncher/main.py [hunk 1]
```diff
@@ -21,7 +20,7 @@ def run_geocruncher(args):
     model = GeologicalModel(args[3])
     model.topography = txt_extract(args[4])
     box = model.getbox()
-    
+
     if args[1] == 'all':
```
Comment (StephenRUK) [inline geocruncher/main.py]: By the way: Usage of arguments should be improved. I have opened a task for this: https://trello.com/c/kQUVpxHE/702-geocruncher-better-handling-of-command-line-arguments

File: geocruncher/MeshGeneration.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        indicator = np.zeros(extended_shape, dtype=np.float32)
+        indicator[1:-1, 1:-1, 1:-1][ranks==rank] = 1
+
+        verts, faces, normals, values = marching_cubes(indicator, level=0.5, gradient_direction="ascent") # Gradient direction ensures normals point outwards
+        tsurf = CGAL.TSurf(rescale(verts, box, shape), faces)
+        bodies.append(tsurf)
+
+    out_files = []
+    for bi, body in enumerate(bodies):
+        # TODO Use unit name from column?
```
Comment (phile314-fh) [inline geocruncher/MeshGeneration.py]: I would not include the unit name, as then we would have to think about special characters etc...

Comment (StephenRUK) [reply geocruncher/MeshGeneration.py]: True, so I will stick with the rank numbers, and we look up the units in the Scala code.

**Comment by StephenRUK** [comment]
FYI @poluxxs if you want to have a look. Perhaps you know how to get the ranks from the GeologicalModel?

**Comment by poluxxs** [comment]
yes, I do know how but only for a given point. 
    model = GeologicalModel(args[3])
    model.topography = txt_extract(args[4])
    model.ranke([x,y,z)])

**Comment by StephenRUK** [comment]
Ok folks, I was able to run this locally with success in interactive mode (importing geocruncher as a module). I don't know if the relative import inside the module works however, I didn't get that to work locally.

**Comment by phile314-fh** [comment]
@StephenRUK Which CGAL/MeshTools version do I need to test this?

**Comment by phile314-fh** [comment]
Can you add a simple test case? Doesn't need to verify the result in depth, but at least check that it does something.

https://github.com/ISSKA/geocruncher/blob/develop/tests/test_geocruncher.py

**Comment by StephenRUK** [comment]
I have MeshTools at the state of commit [0f2e0f11](https://gitlab.inria.fr/charms/MeshTools/tree/0f2e0f11d847321b0938625b7abf13de661f7565)

MeshTools should now support CGAL 4.13 (see [GitLab](https://gitlab.inria.fr/charms/MeshTools/commit/d05c1132b0f23fd0254272485d86ec8a17b79894)). Minimum is 4.12.1 however.

I will add a test case during the morning

**Pull request merged**